content/blog/2025-07-31-valkey-swift/index.md

```diff
@@ -0,0 +1,185 @@
++++
+title= "Introducing valkey-swift, the Swift client for Valkey"
+description = "The new Swift client for Valkey is now in public preview. Learn about its features and how it brings the performance of Valkey to the Swift language."
+date= 2025-08-04 12:00:00
+authors= [ "adamfowler"]
+
+[extra]
+featured = false
++++
+We are excited to introduce the public preview of [valkey-swift](https://github.com/valkey-io/valkey-swift), a new Swift based client library for Valkey.
+
+## The Client
+
+Valkey-swift is a modern swift client built with Swift concurrency in mind. Using Valkey from Swift lets you take advantage of a strongly typed API, Swift's memory and data race safety guarantees, as well as maintaining a very light memory footprint. The API uses structured concurrency; a paradigm designed to bring clarity to concurrent programming by using the structure of your code to define the lifetimes of tasks. The client includes the following features –
```

Comment:
```suggestion
Valkey-swift is a modern swift client built with Swift concurrency in mind. Using Valkey from Swift lets you take advantage of a strongly typed API, Swift's memory and data race safety guarantees, as well as maintaining a very light memory footprint. The API of the client uses structured concurrency; a paradigm that guarantees the resources you need during concurrent tasks are cleaned up when the tasks complete or are cancelled. This allows you to reason about your code locally, rather than worrying about cleaning up after resources the client uses. The client includes the following features –
```

An optional update that adds some detail around why the API using structured concurrency is beneficial to someone adopting this library.